tfhe/src/integer/gpu/server_key/radix/scalar_div_mod.rs

```diff
@@ -157,20 +140,18 @@ impl CudaServerKey {
             .ilog2()
             * numerator.as_ref().d_blocks.lwe_ciphertext_count().0 as u32;
 
-        // Rust has a check on all division, so we shall also have one
         assert_ne!(divisor, Scalar::ZERO, "attempt to divide by 0");
 
         assert!(
             Scalar::BITS >= numerator_bits as usize,
             "The scalar divisor type must have a number of bits that is \
-            >= to the number of bits encrypted in the ciphertext: \n\
-            encrypted bits: {numerator_bits}, scalar bits: {}
-            ",
+        >= to the number of bits encrypted in the ciphertext: \n\
+        encrypted bits: {numerator_bits}, scalar bits: {}
+        ",
             Scalar::BITS
         );
 
         if MiniUnsignedInteger::is_power_of_two(divisor) {
```

Comment:
These checks should be done in the cuda backend as well, based on the value of the divisor the scratch / host should behave differently I think.

Reply:
Same for the check if we should return 0, a boolean can be passed to the scratch & host to determine what to allocate/execute.